src/firebase/api/gcp-utils.js

```diff
@@ -1,74 +1,82 @@
-import { storage, pubsub as gcpPubsub } from '../firebase.js';
-import { pubsub } from '../constants.js';
+import { app, pubsub as gcpPubsub } from '../firebase.js';
+import { pubsub, storageBuckets } from '../constants.js';
 
 /**
  * Upload an image to the specified bucket
  * @param {string} userId userId
  * @param {string} experimentId experimentId
  * @param {Blob} image image file to be uploaded
- * @param {string} bucket destination bucket
  *
  * @return {JSON}
  */
 export const uploadImageToStorage = async (
-  userId, experimentId, image, bucket) => {
+  userId, experimentId, image) => {
   const imagePath = `${userId}-${experimentId}/${image.name}`;
 
-  const storageRef = storage.ref();
-  const rawImageBucketRef = storageRef.child(bucket);
+  const rawImageBucketRef = app.storage(storageBuckets.SIE_RAW_IMGS).ref();
   const newImageRef = rawImageBucketRef.child(imagePath);
-  newImageRef.put(image).then((snapshot) => {
+  return await newImageRef.put(image).then(() => {
     return {
-      status: '201',
-      error: null,
+      status: 201,
+      message: 'image successfully uploaded',
+      data: {},
     };
-  }).catch((error) => {
+  }).catch((err) => {
     return {
-      status: '500',
-      data: null,
-      error,
+      status: 401,
+      message: `not authenticated ${err}`,
+      data: {},
     };
   });
 };
```

Comment:
I'd just make a new function that is specific to sie-experiments which calls this generic `uploadImageToStorage`. If we are hard-coding this, we will eventually have to do this for new experiments with new buckets anyways.

Reply:
ah i highlighted the wrong one. This is meant for the whole `uploadImageToStorage` function.

Reply:
Make sense to me, we can export another wrapper function to the frontend then